applications/nrf_desktop/src/modules/ble_bond.c

```diff
@@ -192,18 +199,31 @@ SETTINGS_STATIC_HANDLER_DEFINE(ble_bond, MODULE_NAME, NULL, settings_set, NULL,
 static uint8_t get_bt_stack_peer_id(uint8_t id)
 {
 	if (IS_ENABLED(CONFIG_BT_PERIPHERAL)) {
-		if ((state == STATE_ERASE_PEER) ||
-		    (state == STATE_ERASE_ADV)) {
+		if ((state == STATE_ERASE_PEER) || (state == STATE_ERASE_ADV) ||
+		(state == STATE_DONGLE_ERASE_PEER) || (state == STATE_DONGLE_ERASE_ADV)) {
 			return bt_stack_id_lut[TEMP_PEER_ID];
 		} else {
-			return bt_stack_id_lut[id];
+			if (state == STATE_DONGLE) {
+				return bt_stack_id_lut[DONGLE_PEER_ID];
+			} else {
+				return bt_stack_id_lut[id];
+			}
```

Comment:
This already is happening with TEMP_PEER_ID. 
How should it behave if you use get_bt_stack_peer_id(tmp_peer_id);
Here I cannot use get_app_id(). | know that we shouldnot be in DONGLE state while we run function with tmp_peer_id but still it can be missileading

Reply:
Discussed offline